### DISCUSSION TITLE
Can test folder contain C code?

### DISCUSSION THREAD

**Discussion by ivan-pi** [fortran-lang/fpm#586] [General]
I have a project, where I would like to have a test driver in C in the `/test` folder.

Fpm still manages to compile the project, however I get warnings like this:

```
 + gfortran -c test/ffttest.c -Wall -Wextra -Wimplicit-interface -fPIC -fmax-errors=1 -g -fcheck=bounds -fcheck=array-temps -fbacktrace -fcoarray=single -J build/gfortran_2A42023B310FA28D/pocketfft-f -I build/gfortran_2A42023B310FA28D/pocketfft-f -I./include -o build/gfortran_2A42023B310FA28D/pocketfft-f/test_ffttest.c.o
cc1: warning: command-line option ‘-Wimplicit-interface’ is valid for Fortran but not for C
cc1: warning: command-line option ‘-fcheck=bounds’ is valid for Fortran but not for C
cc1: warning: command-line option ‘-fcheck=array-temps’ is valid for Fortran but not for C
cc1: warning: command-line option ‘-fbacktrace’ is valid for Fortran but not for C
cc1: warning: command-line option ‘-fcoarray=single’ is valid for Fortran but not for C
```

Should this be considered a bug, or do we want to discourage users from writing tests in C? During development of Fortran libraries which depend on C components, I sometimes find it helpful to first test the C functionality and make sure it works, before I develop the Fortran wrapper.

**Comment by awvwgk** [comment]
I thought I fixed this issue with fpm 0.4.0, at least the latest main branch should default to `gcc` instead of `gfortran` for compiling C code.

However, there is an issue with compiling a C main if I recall correctly: https://github.com/fortran-lang/fpm/discussions/333#discussioncomment-254304

**Reply by ivan-pi** [reply]
Right... I'll come back to this thread after I update to the latest version (currently still running version 0.2.0).
